crates/csharp/src/function.rs

```diff
@@ -225,6 +225,136 @@ impl<'a, 'b> FunctionBindgen<'a, 'b> {
+                payload_is_void = result.ok.is_none();
+                previous = tmp;
+            }
+        }
+        uwriteln!(
+            self.src,
+            "return {};",
+            if payload_is_void { "" } else { &previous }
+        );
+        for (level, var) in vars.iter().enumerate().rev() {
+            self.interface_gen.csharp_gen.needs_wit_exception = true;
+            let (var_name, exception_name) = var;
+            let exception_name = match exception_name {
+                Some(type_name) => &format!("WitException<{}>", type_name),
+                None => "WitException",
+            };
+            uwrite!(
+                self.src,
+                "\
+                                    }} else {{
+                                        throw new {exception_name}({var_name}.AsErr!, {level});
+                                    }}
+                                    "
+            );
+        }
+    }
+
+    fn handle_result_call(
+        &mut self,
+        func: &&wit_parser::Function,
+        target: String,
+        func_name: String,
+        oper: String,
+    ) -> String {
+        let ret = self.locals.tmp("ret");
+        if self.interface_gen.csharp_gen.opts.with_wit_results {
+            uwriteln!(self.src, "var {ret} = {target}.{func_name}({oper});");
+            return ret;
+        }
+
+        // otherwise generate exception code
+        let ty = self
+            .interface_gen
+            .type_name_with_qualifier(func.results.iter_types().next().unwrap(), true);
+        uwriteln!(self.src, "{ty} {ret};");
+        let mut cases = Vec::with_capacity(self.results.len());
+        let mut oks = Vec::with_capacity(self.results.len());
+        let mut payload_is_void = false;
+        for (index, ty) in self.results.iter().enumerate() {
+            let TypeDefKind::Result(result) = &self.interface_gen.resolve.types[*ty].kind else {
+                unreachable!();
+            };
+            let err_ty = if let Some(ty) = result.err {
+                self.interface_gen.type_name_with_qualifier(&ty, true)
+            } else {
+                "None".to_owned()
+            };
+            let ty = self
+                .interface_gen
+                .type_name_with_qualifier(&Type::Id(*ty), true);
+            let head = oks.concat();
+            let tail = oks.iter().map(|_| ")").collect::<Vec<_>>().concat();
+            cases.push(format!(
+                "\
+                                            case {index}: {{
+                                                ret = {head}{ty}.Err(({err_ty}) e.Value){tail};
+                                                break;
+                                            }}
+                                            "
+            ));
+            oks.push(format!("{ty}.Ok("));
+            payload_is_void = result.ok.is_none();
+        }
+        if !self.results.is_empty() {
+            self.src.push_str("try {\n");
```

Comment:
```suggestion
            self.src.push_str(\"
                                      try 
                                      {\n");
```
Not sure I've got the rust syntax right here.